cmake/options.cmake

```diff
@@ -17,6 +17,8 @@ option(BUILD_SHARED_LIBS "Whether to build shared libraries (like *.dll or *.so)
 
 option(ADD_HEADERS_AS_SOURCES "Whether to add headers as sources of a target or not. This is needed for some IDEs which wouldn't detect headers properly otherwise")
 
+option(BUILD_TESTING "Whether to enable and build tests or not")
```

Comment:
Should this also have a default of `OFF`?

Reply:
If the starting value is not specified, the default is OFF.

Reply:
I like the implicitness of having a default

Reply:
> I like the implicitness of having a default

Do you mean explicitness?

Reply:
hahahaha. yes.

Reply:
I thought it would've been redundant since you don't have multiple defaults to take into account.
That been said, I'll merge this and do those changes in another PR, where I'm doing some small cleanups to variables.